pkg/storage/tsdb/config.go

```diff
@@ -48,31 +48,38 @@ type Config struct {
 // DurationList is the block ranges for a tsdb
 type DurationList []time.Duration
 
-// String implements the flag.Var interface
-func (d DurationList) String() string { return "RangeList is the block ranges for a tsdb" }
-
-// Set implements the flag.Var interface
-func (d DurationList) Set(s string) error {
-	blocks := strings.Split(s, ",")
-	d = make([]time.Duration, 0, len(blocks)) // flag.Parse may be called twice, so overwrite instead of append
-	for _, blk := range blocks {
-		t, err := time.ParseDuration(blk)
+// String implements the flag.Value interface
+func (d *DurationList) String() string {
+	values := make([]string, 0, len(*d))
+	for _, v := range *d {
+		values = append(values, v.String())
+	}
+
+	return strings.Join(values, ",")
+}
+
+// Set implements the flag.Value interface
+func (d *DurationList) Set(s string) error {
+	values := strings.Split(s, ",")
+	*d = make([]time.Duration, 0, len(values)) // flag.Parse may be called twice, so overwrite instead of append
+	for _, v := range values {
+		t, err := time.ParseDuration(v)
 		if err != nil {
 			return err
 		}
-		d = append(d, t)
+		*d = append(*d, t)
 	}
 	return nil
 }
 
-// ToMillisecondRanges returns the duration list in milliseconds
-func (d DurationList) ToMillisecondRanges() []int64 {
-	ranges := make([]int64, 0, len(d))
-	for _, t := range d {
-		ranges = append(ranges, int64(t/time.Millisecond))
+// ToMillisecond returns the duration list in milliseconds
```

Comment:
nit: Should be plural? It's not just one millisecond, it's slice of milliseconds.